lib/core/src/main/java/io/atlasmap/core/DefaultAtlasFunctionResolver.java

```diff
@@ -44,17 +49,51 @@ private void init() {
         for (FunctionFactory f : implementations) {
             functions.put(f.getName().toUpperCase(), f);
         }
+
+        fieldActionService = new DefaultAtlasFieldActionService(DefaultAtlasConversionService.getInstance());
+        fieldActionService.init();
```

Comment:
Is there any reason why DefaultAtlasFieldActionService is not a singleton like e.g. DefaultAtlasFunctionResolver or DefaultAtlasConversionService...?

Reply:
Nothing I'm aware of, seems better to be a singleton